### ISSUE TITLE
v4.2 from 4.1: cancelled effects no longer provide access to the cancelled function name

### ISSUE THREAD

**Issue by rplotkin** [jfairbank/redux-saga-test-plan#96]
Previously working code
```
let myFunctionWasCancelled = false
return expectSaga(watchStartChat)
.provide({
  cancel: (effect, next) => {
    if (effect.name === 'myFunction') {
      myFunctionWasCancelled = true
    }
    return next()
  }
})
...
```
effect.name used to be the name of the function being cancelled, but now `effect.name === 'sagaWrapper'`
there doesn't seem to be another part of the object from which I can extract the function name

**Comment by jfairbank** [comment]
Thanks for opening this issue @rplotkin!

Interestingly, I'm seeing this behavior in older versions as well. Regardless, I'll work on a bug fix.

**Comment by jfairbank** [comment]
@rplotkin I've pushed a fix in v2.4.2.

Please let me know if you have any more issues.
